lib/rewrite-middleware.js

```diff
@@ -10,18 +11,28 @@ let {nonIncludedFiles, REVERSE_CONTEXT} = require('./framework.js');
 // Replicate filesPromise functionality from webServer.
 // Unfortunately we can’t access the webServer child DI injector
 // from here where the filesPromise lives.
-function createFilesPromise (fileList, emitter) {
+function createFilesPromise(fileList, emitter) {
 	// Set an empty list of files to avoid race issues with
 	// file_list_modified not having been emitted yet
-	let files = fileList.files
-	emitter.on('file_list_modified', (filesParam) => { files = filesParam })
-  
-	return {
-	  then (...args) {
-		return Promise.resolve(files).then(...args)
-	  }
+	if (KARMA_MAJOR_VERSION < 5) {
+		var filesPromise = new common.PromiseContainer()
+		filesPromise.set(Promise.resolve(fileList.files));
+		emitter.on('file_list_modified', function (files) {
+			filesPromise.set(Promise.resolve(files))
+		});
+		return filesPromise;
+
+	} else {
+		let files = fileList.files
+		emitter.on('file_list_modified', (filesParam) => { files = filesParam })
+
+		return {
+			then(...args) {
+				return Promise.resolve(files).then(...args)
+			}
+		}
```

Comment:
Wouldn’t the following also work:

```suggestion
		return new Promise(resolve => {
			emitter.on('file_list_modified', resolve);
		});
```

?

Reply:
I don't think so. As before, the promise functionality was replicated karma webServer.

Reply:
Ah, yes, I see